src/BlockTemplatesController.php

```diff
@@ -360,4 +363,15 @@ public function render_block_template() {
 		}
 	}
 
+	/**
+	 * When the search is for products and a block theme is active, render the Product Search Template.
+	 *
+	 * @param array $templates Templates that match the search hierarchy.
+	 */
+	public function update_search_template_hierarchy( $templates ) {
```

Comment:
Just going to mention this comment by @gigitux here again because we should avoid convoluting this controller with very specific template logic such as the search page.

https://github.com/woocommerce/woocommerce-gutenberg-products-block/pull/6345/files#r861980598